SYCL/ESIMD/api/functional/shared_element.hpp

```diff
@@ -0,0 +1,50 @@
+//===-- shared_elements.hpp - Function that provides USM with a smart pointer.
+//      -------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+///
+/// \file
+/// This file provides data struct that lets interact with USM with a smart
+/// pointer that lets avoid memory leaks.
+///
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <functional>
+#include <memory>
+#include <sycl/sycl.hpp>
+
+namespace esimd_test::api::functional {
+
+template <typename T> class shared_element {
```

Comment:
please add some docs how this struct achieves the stated goals

Reply:
I've applied this comment [here](https://github.com/intel/llvm-test-suite/pull/762/commits/f6c0e9809f2bbdef3c23496b8f2dd8e8c5e07a88#diff-9d2d5320f066e81e77b17efb5f2888df4d6d5e653f591eb2e9d39f34c267172cR25-R27)